float32 y_max
float32 y_min
